src/SubscriberBase.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/MoveWrapper.h>
+#include "src/EnableSharedFromThis.h"
+#include "src/ReactiveStreamsCompat.h"
+#include "src/mixins/ExecutorMixin.h"
+
+namespace reactivesocket {
+
+template <typename T>
+class SubscriberBaseT : public Subscriber<T>,
+                        public EnableSharedFromThisBase<SubscriberBaseT<T>>,
+                        public virtual ExecutorBase {
+  virtual void onSubscribeImpl(std::shared_ptr<Subscription> subscription) = 0;
```

Comment:
Is default visibility `protected`? I didn't realize that—maybe worth adding it explicitly?